File: ButterKnifeDemo/app/src/main/java/com/javahelps/butterknifedemo/MainActivity.java
package com.javahelps.butterknifedemo;

import android.os.Bundle;
import android.support.v7.app.ActionBarActivity;
import android.widget.EditText;
import android.widget.Toast;

import butterknife.ButterKnife;
import butterknife.InjectView;
import butterknife.OnClick;


public class MainActivity extends ActionBarActivity {
    @InjectView(R.id.etName)
    EditText etName;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ButterKnife.inject(this);
    }

    @OnClick(R.id.btnOK)
    public void showName() {
        String name = etName.getText().toString();
        Toast.makeText(this, name, Toast.LENGTH_SHORT).show();
    }
}

File: kotlin/HelloWorld/app/src/main/java/com/javahelps/helloworld/MainActivity.kt
package com.javahelps.helloworld

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.Button
import android.widget.EditText
import android.widget.Toast

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // Find the EditText etName
        var etName = findViewById<EditText>(R.id.etName)
        // Find the Button btnSayHello
        var btnSayHello = findViewById<Button>(R.id.btnSayHello);

        // Set onclick listener
        btnSayHello.setOnClickListener {
            Toast.makeText(this, "Hello ${etName.text}", Toast.LENGTH_LONG).show();
        }
    }
}
